enhancements/machine-api/out-of-tree-provider-support.md

```diff
@@ -0,0 +1,474 @@
+secret-namespace = kube-system
+```
+
+Resources deployed here will be destroyed with the bootstrap stage cleanup.
+
+#### Operator resource management strategy
+
+Every cloud manages it's configuration differently, but shares a common overall structure.
+
+1. `cloud-controller-manager` and related resources are deployed under a single namespace.
+2. Each pod (`CCM`, `cloud-node-manager`, etc.) will be managed by a `Deployment`.
+3. The pods are expected to run on the control-plane `Nodes` using `hostNetwork`.
+
+Operator will manage:
+
+- Static infrastructure resources creation, such as namespaces or service accounts will be delegated to the [CVO](https://github.com/openshift/cluster-version-operator).
+- Operator will populate, create and manage the cloud-specific set of `Deployments`, based on the selected `platform` in the `Infrastructure` resource.
+- `Deployments`, provisioned by the operator will run the pods on the `control-plane` nodes, and will tolerate `cloudprovider.kubernetes.io/uninitialized` and `node-role.kubernetes.io/master` taints.
+- The `--cloud-provider` and the `--cloud-config` arguments will be populated using existing `library-go` based implementation from the [cluster-kube-controller-manager-operator](https://github.com/openshift/cluster-kube-controller-manager-operator/blob/990ea3f2ace13be6579c00a889842c3f5d3e756a/pkg/operator/configobservation/configobservercontroller/observe_config_controller.go#L82-L85) to share a common approach with the bootstrap specific static-pod configuration.
+- Operator will own a `cluster-operator` resource, which will report the readiness of all workloads at post-install phase. The conditions reported will help other dependent components, such as [openshift-ingress](https://github.com/openshift/cluster-ingress-operator).
+- Operator will be built with a goal to achieve provider decoupling and potential switching between providers in the cluster, and will consider support to run multiple providers in a cluster simultaneously (long-term goal).
+
+#### CVO managemnet
+
+The operator image will be build and included in every release payload, and expected to be deployed and running at kuberentes operators level ([10-29](https://github.com/openshift/cluster-version-operator/blob/34010292c3abd2582b687e9c0ef76c5924998f39/docs/dev/operators.md#how-do-i-get-added-as-a-special-run-level)) right after the `kube-controller-manager-operator` at runlevel [25](https://github.com/openshift/cluster-kube-controller-manager-operator/tree/master/manifests).
+
+Proposed `CCMO` manifests runlevel is `26`.
+
+#### Cloud provider management
+
+Each cloud provider will be hosted under its own repository. Those repositories will be initially bootstrapped by forking upstream ones. 
+
+The `CVO` will be responsible for provisioning static resources, such as `SA`, `RBAC`, `ServiceMonitors` for metrics for the operator. Then, the operator will  be responsible for constructing and creating cloud provider resources.
```

Comment:
Should spell out `SA` as `ServiceAccounts` and `RBAC` as `Roles/RoleBindings`. No real need to abbreviate here IMO